backend/btrixcloud/orgs.py

```diff
@@ -801,27 +802,53 @@ async def get_orgs(
 
     @app.post("/orgs/create", tags=["organizations"])
     async def create_org(
-        new_org: CreateOrg,
-        user: User = Depends(user_dep),
+        new_org: OrgCreate,
+        request: Request,
+        user: User = Depends(user_or_shared_secret_dep),
     ):
         if not user.is_superuser:
             raise HTTPException(status_code=403, detail="Not Allowed")
 
         id_ = uuid4()
 
+        name = new_org.name or str(id_)
+
         if new_org.slug:
             validate_slug(new_org.slug)
             slug = new_org.slug
         else:
-            slug = slug_from_name(new_org.name)
+            slug = slug_from_name(name)
 
         org = Organization(
-            id=id_, name=new_org.name, slug=slug, users={}, storage=ops.default_primary
+            id=id_,
+            name=name,
+            slug=slug,
+            users={},
+            storage=ops.default_primary,
+            quotas=new_org.quotas or OrgQuotas(),
+            subData=new_org.subData,
         )
         if not await ops.add_org(org):
             return {"added": False, "error": "already_exists"}
 
-        return {"id": id_, "added": True}
+        result = {"added": True, "id": id_}
+
+        if new_org.firstAdminInviteEmail:
+            if await invites.invite_user(
+                InviteToOrgRequest(
+                    email=new_org.firstAdminInviteEmail, role=UserRole.OWNER
+                ),
+                user,
+                user_manager,
+                org=org,
+                allow_existing=True,
+                headers=request.headers,
+            ):
+                result["invited"] = "new_user"
+            else:
+                result["invited"] = "existing_user"
```

Comment:
In this case are we actually inviting an existing user yet or just creating the org without sending any invitations yet at this stage? I think the latter, yeah? In that case it might make sense to just omit this or set the value to `None`?

Reply:
yes, `invited` is set if `new_user` or `existing_user` is invited, otherwise it will be unset (undefined in response).

Reply:
this is just reusing the return values from the invite endpoint, with possibility of it being unset as well.